sysconfig/cloudinit.go

```diff
@@ -112,12 +112,18 @@ func configureCloudInit(opts *Options) (err error) {
 
 	// next check if there is a gadget cloud.conf to install
 	if HasGadgetCloudConf(opts.GadgetDir) {
-		// then copy / install the gadget config and return without considering
-		// CloudInitSrcDir
-		// TODO:UC20: we may eventually want to consider both CloudInitSrcDir
-		// and the gadget cloud.conf so returning here may be wrong
+		// then copy / install the gadget config first
 		gadgetCloudConf := filepath.Join(opts.GadgetDir, "cloud.conf")
-		return installGadgetCloudInitCfg(gadgetCloudConf, WritableDefaultsDir(opts.TargetRootDir))
+		err := installGadgetCloudInitCfg(gadgetCloudConf, WritableDefaultsDir(opts.TargetRootDir))
```

Comment:
Got me thinking a bit, I know that we use osutil.CopyFile without the overwrite flag. Perhaps we should add a test that demonstrates that if the gadget has a file and ubuntu-seed ships a file named `80_device_gadget.cfg` configureCloudInit() will fail with EEXIST?